src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -874,6 +874,32 @@ instruct vfmsub_masked(vReg dst_src1, vReg src2, vReg src3, vRegMask_V0 v0) %{
   ins_pipe(pipe_slow);
 %}
 
+// MulAddVS2VI
+// Vector Multiply-Add Shorts into Integer
+// s1[0]*s2[0] + s1[1]*s2[1] => int[0]
+
+instruct vmuladdS2I(vReg dst, vReg src1, vReg src2, vReg tmp1, vReg tmp2, vReg tmp3) %{
+  predicate(Matcher::vector_element_basic_type(n->in(1)) == T_SHORT);
+  match(Set dst (MulAddVS2VI src1 src2));
+  effect(TEMP_DEF dst, TEMP tmp1, TEMP tmp2, TEMP tmp3);
+  format %{ "vmuladdS2I $dst, $src1, $src2\t" %}
+  ins_encode %{
+    __ vsetvli_helper(T_INT, Matcher::vector_length(this));
+
+    __ vsll_vi(as_VectorRegister($dst$$reg), as_VectorRegister($src1$$reg), 16);
+    __ vsra_vi(as_VectorRegister($dst$$reg), as_VectorRegister($dst$$reg), 16);
+    __ vsra_vi(as_VectorRegister($tmp1$$reg), as_VectorRegister($src1$$reg), 16);
+
+    __ vsll_vi(as_VectorRegister($tmp2$$reg), as_VectorRegister($src2$$reg), 16);
+    __ vsra_vi(as_VectorRegister($tmp2$$reg), as_VectorRegister($tmp2$$reg), 16);
+    __ vsra_vi(as_VectorRegister($tmp3$$reg), as_VectorRegister($src2$$reg), 16);
+
+    __ vmul_vv(as_VectorRegister($dst$$reg), as_VectorRegister($dst$$reg), as_VectorRegister($tmp2$$reg));
+    __ vmacc_vv(as_VectorRegister($dst$$reg), as_VectorRegister($tmp1$$reg), as_VectorRegister($tmp3$$reg));
```

Comment:
Hmm ... This doesn't look like a simple sequence, isn't it? It's hard to tell whether we will benifit from this change without JMH testing on real RVV hardwares especially when VLEN is not large (No big difference in respect of number of instructions executed when VLEN=128-bits).

Reply:
You're right.
I'm waiting for my board to test it. I'll update when I get the data.

Reply:
The general advantage over no intrinsic is that we avoid the spilling from the vector register to the heap in case it falls back to whatever the C2 compiler would compile to. That avoids hit to the L1 or load/store CPU pipe even. I agree, we'll need runs with JMH on actual hardware to verify that it's indeed a win.